### ISSUE TITLE
[stable/datadog] custom commands in the init container ? 

### ISSUE THREAD

**Issue by bkruger99** [helm/charts#22468]
I am making a proposal for the ability to execute custom commands in the init container to be able to install additional packages.

For example, I would like to monitor GPU across my GPU hosts.   There is nothing stock in the agent.  Someone has written checks here:  https://github.com/ngi644/datadog_nvml  

This requires a pip install and copying files over.  I know there is a mechanism to do inline, but that doesn't feel like a great method to me.  Especially inside of kubernetes and multiple custom checks if one chooses.  Something about inline python in a yaml file just feels dirty. 

**Describe the solution you'd like**
It would be nice to be able to take the chart, add a custom init section, and give a list of commands to execute.   This would need to be in the init_container to put everything into place prior to the agent starts.  This would offer up similar functionality to running the agent outright on the host.  This would be executed last.  It could also be a second init container to fully isolate.  I'm not picky. :) 

**Describe alternatives you've considered**
Pre-pending this on my own to your chart after I write it and deploying it locally.
Forking the container and building it on my own with a nice method to install the needful.

**Additional context**
I will likely be going down the road of pre-pending on the chart, but I would like to gauge what the interest level of a pull request here is and an acceptance.  I would like to contribute upstream and not have to build/patch/etc. every time there is a new chart release.  I will happily build this because I need a solution anyway, so any guidance as to getting this merged into the chart would be graciously accepted.

Unfortunately, I don't know windows PowerShell enough to be able to fix the windows config.
